src/App.tsx

```diff
@@ -1,61 +1,139 @@
 import './App.scss';
+import { useState } from 'react';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find((user) => user.id === userId);
+
+  return foundUser || null;
+}
+
+const initialTodos = todosFromServer.map((todo) => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [todos, setTodos] = useState(initialTodos);
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [errorTitle, showErrorTitle] = useState(false);
+  const [errorUser, showErrorUser] = useState(false);
+  const [title, setTitle] = useState('');
+
+  const handleClear = () => {
+    setTitle('');
+    setSelectedUserId(0);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!title.trim()) {
+      showErrorTitle(true);
+    }
+
+    if (!selectedUserId) {
+      showErrorUser(true);
+    }
+
+    if (title.trim() && selectedUserId) {
+      const newTodo: Todo = {
+        id: Math.max(...todos.map((todo) => todo.id)) + 1,
+        userId: selectedUserId,
+        title,
+        completed: false,
+        user: getUser(selectedUserId),
+      };
+
+      setTodos((prevTodo => [...prevTodo, newTodo]));
+
+      handleClear();
+    }
+  };
+
+  const handleTittle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    setTitle(value);
+    showErrorTitle(false);
+
+    setTitle(value.replace(/[^A-Za-z\s\d\u0400-\u04FF]/g, ''));
+  };
+
+  const handleUser = (event: React.ChangeEvent<HTMLSelectElement>) => {
```

Comment:
Consider using a more descriptive name for `handleUser`. For example, `handleUserSelection`. 🤔